features/nanostack/FEATURE_NANOSTACK/nanostack_entropy_check.cpp

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2015 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbedtls/platform.h"
+
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

Comment:
None of that makes any sense to me. The requirement of the code is that SSL be available. Simple as that. The reason why it's not available is not of concern.

So the test is just MBEDTLS_SSL_TLS_C.

And including "mbedtls/config.h" doesn't work, as it won't necessarily be the config in use. If testing that flag, it would make sense to be including "mbedtls/ssl.h", as that should logically be pulling in the actual config file to decide what to provide.

(Although it actually provides all prototypes even if SSL is disabled).